### PR TITLE
feat(pipeline): Optimize pipeline db gc

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Optimize pipeline db gc

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTQ2XSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=295146&iterationID=1146&pId=290591&type=TASK)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      Optimize pipeline db gc        |
| 🇨🇳 中文    |       优化pipeline db gc       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/dbgc/dbgc.go [hunk 1]
```diff
@@ -77,18 +77,19 @@ func (p *provider) doPipelineDatabaseGC(req apistructs.PipelinePageListRequest)
...
-			}
+		pipelines = append(pipelines, pipelineResults...)
+	}
+	for _, pipeline := range pipelines {
+		if !pipeline.Status.CanDelete() {
+			continue
+		}
+		// gc logic
+		if err := p.DoDBGC(pipeline.PipelineID, apistructs.PipelineGCDBOption{NeedArchive: needArchive(pipeline)}); err != nil {
+			logrus.Errorf("failed to do gc logic, pipelineID: %d, err: %v", pipeline.PipelineID, err)
```
Comment (sfwn) [inline modules/pipeline/providers/dbgc/dbgc.go:90]: use p.Log

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
it's not graceful: modules/pipeline/providers/dbgc/dbgc.go:44

**Comment by sfwn** [comment]
make modules/pipeline/providers/dbgc/dbgc.go:50 configurable, and others.

File: modules/pipeline/providers/dbgc/dbgc.go (hunk 1 above)
Comment (littlejiancc) [reply modules/pipeline/providers/dbgc/dbgc.go]: done

File: modules/pipeline/conf/conf.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ type Conf struct {
 	// build cache
 	BuildCacheCleanJobCron string        `env:"BUILD_CACHE_CLEAN_JOB_CRON" default:"0 0 0 * * ?"`
 	BuildCacheExpireIn     time.Duration `env:"BUILD_CACHE_EXPIRE_IN" default:"168h"`
+	PipelineDBGCCron       string        `env:"PIPELINE_DBGC_CRON" default:"0 0 0/2 * * ?"`
```
Comment (sfwn) [inline modules/pipeline/conf/conf.go:47]: move to provider self config.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**